practical/swarm_demo/install.sh

```diff
@@ -5,19 +5,28 @@
 #
 # USAGE: chmod +x ./install.sh && ./install.sh
 #
-# DESCRIPTION: Tears down an existing Swarm hosts and
-#              provisions a fresh cluster.
-
+# DESCRIPTION: Creates a Docker Swarm cluster with a discovery service node (Consul),
+#              a master and two slave nodes and runs the CA674 demo app on top of it.
+#              The CA674 demo is comprised of a back-end store (redis) and front-end
+#              www implementation (python) scaled to initially two instances.
+#              A load balancer is used to randomize access to each front-end container
+#              and unify the entry point of the app.
+#
 #===================================================================================
 
-VM=local
+function teardown_swarm_machines {
+  echo "Removing old Swarm machines..."
+  docker-machine stop keystore
+  docker-machine stop swarm-master
+  docker-machine stop swarm-node-01
+  docker-machine stop swarm-node-02
+  docker-machine rm -f keystore
+  docker-machine rm -f swarm-master
+  docker-machine rm -f swarm-node-01
+  docker-machine rm -f swarm-node-02
+}
 
 function provision_machines {
-    echo "Entering $VM machine..."
-    docker-machine status $VM || docker-machine create -d virtualbox $VM && sleep 2
-    docker-machine start $VM && sleep 2
-    eval "$(docker-machine env $VM)"
-
     echo "Creating key-value store machine..."
     docker-machine create -d virtualbox keystore
```

Comment:
Similar to what I mentioned above, this will break if `keystore` already exists.
Best to stick with the old approach `docker-machine status $VM || docker-machine create -d virtualbox $VM`